sysl2/sysl/binexprEval.go

```diff
@@ -113,9 +113,16 @@ func (op Eval_LHS_Over_RhsStrategy) eval(txApp *sysl.Application, assign *Scope,
 	vType := getValueType(lhs_v)
 	itemType := getContainedType(lhs_v)
 	key := makeKey(binexpr.Op, vType, itemType)
-
+	scopeVarValue, hasScopeVar := (*assign)[binexpr.Scopevar]
```

Comment:
`assign` doesn't need to be a `*Scope`. It can be a `Scope`. Maps are already reference types.